client/components/blogging-prompt/index.jsx

```diff
@@ -8,47 +8,61 @@ import { useBloggingPrompts } from 'calypso/data/blogging-prompt/use-blogging-pr
 import useSkipCurrentViewMutation from 'calypso/data/home/use-skip-current-view-mutation';
 import { SECTION_BLOGGING_PROMPT } from 'calypso/my-sites/customer-home/cards/constants';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
-import getPrimarySiteId from 'calypso/state/selectors/get-primary-site-id';
-import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import getSiteSlug from 'calypso/state/sites/selectors/get-site-slug';
 import BellOffIcon from './bell-off-icon';
 import LightbulbIcon from './lightbulb-icon';
 import PromptsNavigation from './prompts-navigation';
 
 import './style.scss';
 
-const BloggingPromptCard = () => {
+const BloggingPromptCard = ( { siteId, viewContext } ) => {
 	const dispatch = useDispatch();
 	const translate = useTranslate();
-	const selectedSiteId = useSelector( ( state ) => getSelectedSiteId( state ) );
-	const primarySiteId = useSelector( ( state ) => getPrimarySiteId( state ) );
-	let siteId = selectedSiteId;
-
-	// We need a site ID to request prompts
-	// If no selected site ID, fallback to using Primary site ID
-	if ( siteId === null ) {
-		siteId = primarySiteId;
-	}
-
-	const siteSlug = useSelector( ( state ) => getSelectedSiteSlug( state ) );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
 	const notificationSettingsLink = '/me/notifications' + ( siteSlug ? '#' + siteSlug : '' );
-
 	const maxNumberOfPrompts = 10;
 	const { data: prompts } = useBloggingPrompts( siteId, maxNumberOfPrompts );
-
 	const { skipCard } = useSkipCurrentViewMutation( siteId );
 
 	if ( prompts === undefined ) {
 		return null;
 	}
+
+	if ( viewContext !== 'reader' ) {
+		viewContext = 'home';
+	}
```

Comment:
This doesn't seem like it should be needed? Can we pass the viewContext prop in from /home, as well?

Reply:
We can. I was reluctant as this was the only card on the home page that used it however. I'll make this change and see how it looks.

Reply:
Taken out